source/numeric/ConstantValue.cpp

```diff
@@ -360,6 +361,45 @@ ConstantValue ConstantValue::convertToStr() const {
     return result;
 }
 
+ConstantValue ConstantValue::convertToByteArray(bitwidth_t size, bool isSigned) const {
+    if (isUnpacked())
+        return *this;
+    std::string result;
+    if (isInteger()) {
```

Comment:
Yes, inside the new convertToByteArray I will replace copied code with one line
```c++
        result = convertToStr().str();
```

Reply:
You have to check validity first; if evaluation fails it will return a ConstantValue that is "bad" and you need to just return a bad value in that case (return nullptr). But otherwise yeah, that works.

Reply:
My code checks isInteger() before calling convertToStr()
```c++
    if (isInteger()) {
        result = convertToStr().str();
    }
```
and the current convertToStr() code cannot fail if isInteger() is true. Do you still think another check on convertToStr() return value is necessary?

Reply:
Ah yeah, if you're already checking isInteger() then it's fine.